src/Raven.Server/Documents/ETL/Providers/Raven/RavenEtl.cs

```diff
@@ -32,7 +33,19 @@ public RavenEtl(Transformation transformation, RavenEtlConfiguration configurati
             if (configuration.TestMode == false)
                 _requestExecutor = RequestExecutor.Create(configuration.Connection.TopologyDiscoveryUrls, configuration.Connection.Database, serverStore.Server.Certificate.Certificate, DocumentConventions.Default);
             
-            _script = new RavenEtlDocumentTransformer.ScriptInput(transformation);            
+            _script = new RavenEtlDocumentTransformer.ScriptInput(transformation);         
+			
+			serverStore.Server.ServerCertificateChanged += (sender, args) =>
+            {
+                // When the server certificate changes, we need to start using the new one.
+                // Since the request executor has the old certificate, we will re-create it and it will pick up the new certificate.
+                var newRequestExecutor = RequestExecutor.Create(configuration.Connection.TopologyDiscoveryUrls, configuration.Connection.Database, serverStore.Server.Certificate.Certificate, DocumentConventions.Default);
```

Comment:
wrap this into a method and use this in both places. You will avoid a situation where someone will change the initial creation of the executor and forget about the event.

Reply:
OK